src/cow-react/modules/swap/services/ethFlow/steps/analytics.ts

```diff
@@ -0,0 +1,41 @@
+import { signSwapAnalytics, swapAnalytics } from 'components/analytics'
+import TradeGp from 'state/swap/TradeGp'
+import { USER_SWAP_REJECTED_ERROR } from '@cow/modules/swap/services/common/steps/swapErrorHelper'
+
+function getMarketLabel(trade: TradeGp): string {
+  return [trade.inputAmount.currency.symbol, trade.outputAmount.currency.symbol].join(',')
+}
+
+export interface SwapFlowAnalyticsContext {
+  account: string | null
+  recipient: string | null
+  recipientAddress: string | null
+  trade: TradeGp
+}
+
+export const swapFlowAnalytics = {
+  swap(context: SwapFlowAnalyticsContext) {
+    swapAnalytics('Send', getMarketLabel(context.trade))
+  },
+  sign(context: SwapFlowAnalyticsContext) {
+    const { account, recipient, recipientAddress, trade } = context
+    const marketLabel = getMarketLabel(trade)
+
+    if (recipient === null) {
+      signSwapAnalytics('Sign', marketLabel)
+    } else {
+      ;(recipientAddress ?? recipient) === account
```

Comment:
Why do you need the `;(...)` here?

Reply:
it's a set `var` already and checking via the `??` makes that a formatting requirement